# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1337367395 25200
#      Fri May 18 11:56:35 2012 -0700
# Node ID b9151573a23045aa3ea79ef6e406607ee113a731
# Parent  f19d29868346e63145c7849c59b2d896d3f5649e
build: Allow VPATH builds of AUTHORS file.

* interpreter/Makefile.am: Copy contributors.texi file from $srcdir to
current build directory if necessary.

diff --git a/doc/interpreter/Makefile.am b/doc/interpreter/Makefile.am
--- a/doc/interpreter/Makefile.am
+++ b/doc/interpreter/Makefile.am
@@ -193,16 +193,20 @@ doc-cache: $(DOCSTRING_FILES) mk_doc_cac
 	mv $@-t $@
 
 contributors.texi: contributors.in
 	$(AWK) -f $(srcdir)/mkcontrib.awk $(srcdir)/contributors.in > $@-t
 	mv $@-t $@
 
 ../../AUTHORS: preface.texi contributors.texi
 	rm -f AUTHORS
+	if [ "x$(srcdir)" != "x." ] && [ -f $(srcdir)/contributors.texi ] && [ ! -f contributors.texi ]; then \
+		cp $(srcdir)/contributors.texi contributors.texi; \
+		touch -r $(srcdir)/contributors.texi contributors.texi; \
+	fi
 	-$(MAKEINFO) -D AUTHORSONLY \
 	  --no-validate --no-headers --no-split --output AUTHORS $<
 	mv AUTHORS ../../AUTHORS
 
 ../../BUGS: bugs.texi
 	rm -f BUGS
 	-$(MAKEINFO) -D BUGSONLY \
 	  --no-validate --no-headers --no-split --output BUGS $<
